src/output/mod.rs

```diff
@@ -40,8 +40,15 @@ pub async fn log(config: Config) {
 
     let mut interval = tokio::time::interval_at(start, config.general().interval());
 
-    while Instant::now() + config.general().interval() <= stop {
-        interval.tick().await;
+    while RUNNING.load(Ordering::Relaxed) && Instant::now() + config.general().interval() <= stop {
+        // use a timeout here so we always check RUNNING at least once a second
+        if timeout(Duration::from_secs(1), interval.tick())
```

Comment:
Will this end up misaligning our metrics (in time) if the interval is not a whole number of seconds? Or does the interval keep state internally of when it is supposed to tick and not restart the whole wait if it times out?

(I think it does the latter but wanted to confirm).

Reply:
No. The interval keeps state internally so there is no issue with other intervals.